app/views/questionnaire.py

```diff
@@ -272,15 +267,16 @@ def is_survey_completed(answer_store, metadata):
 def _save_sign_out(collection_id, eq_id, form_type, this_location, form):
     questionnaire_store = get_questionnaire_store(current_user.user_id, current_user.user_ik)
     block_json = _render_schema(this_location)
-
     if _is_invalid_form(form):
         content = {'block': block_json, 'form': form}
         return _build_template(this_location, content, template=block_json['type'])
     else:
         _update_questionnaire_store(this_location, form)
-
         if this_location in questionnaire_store.completed_blocks:
             questionnaire_store.completed_blocks.remove(this_location)
+            questionnaire_store.add_or_update()
+        session_storage.clear()
```

Comment:
personal preference but a line break after the indentation makes it look clearer that the if has finished?